addons/mail/models/mail_thread.py

```diff
@@ -1337,16 +1342,17 @@ def message_route_process(self, message, message_dict, routes):
             # replies to internal message are considered as notes, but parent message
             # author is added in recipients to ensure he is notified of a private answer
             partner_ids = []
-            if message_dict.pop('internal', False):
-                subtype = 'mail.mt_note'
-                if message_dict.get('parent_id'):
-                    parent_message = self.env['mail.message'].sudo().browse(message_dict['parent_id'])
-                    if parent_message.author_id:
-                        partner_ids = [(4, parent_message.author_id.id)]
-            else:
-                subtype = 'mail.mt_comment'
+            if not subtype_id:
```

Comment:
Don't forget to add in pad functional change that now incoming email has subtype (impact for example lead users).